report/src/sections/03-preliminaries.tex

```diff
@@ -100,4 +100,31 @@ \subsection{Converting Code Program Graph}\label{subsection:converting-code-prog
     \input{figures/program-graph}
     \caption{An example that shows a program graph of an assignment}
     \label{fig:tikz-program-graph-assignment}
-\end{figure}    
\ No newline at end of file
+\end{figure}
+
+We later describe the abstract syntax; there will be no control structures, like if-statements and loops.
+This is because the abstract syntax is simplified to make understanding the theory behind abstract interpretation easier.
+if-statements and loops are still used as boolean expressions in edges, used like guards in the program graph.
+In \autoref{fig:tikz-program-graph-loop} and \autoref{fig:tikz-program-graph-if}, we see examples of how a loop and an if-statement are represented in a program graph.
+We say $b$ is some Boolean expression, where $S$ is a set of states done in the control structure.
+In the abstract syntax, we will not have formulated sequences of $I_0, I_1, I_2, \dots, I_n$ as in the program graph, but rather as a set of states $S$, where we at the end go back to the initial state, this is shown in \autoref{fig:tikz-composition-graph}.
+\begin{figure}[htb!]
+    \center
+    \input{figures/loop_graph}
+    \caption{An example that shows a program graph of a loop}
+    \label{fig:tikz-program-graph-loop}
+\end{figure}
+
+\begin{figure}[htb!]
+    \center
+    \input{figures/if_graph}
+    \caption{An example that shows a program graph of an if-statement}
+    \label{fig:tikz-program-graph-if}
+\end{figure}
+
+\begin{figure}
+    \center
+    \input{figures/composition_graph}
+    \caption{An example that shows a composition graph}
+    \label{fig:tikz-composition-graph}
+\end{figure}
```

Comment:
Måske burde vi vise den abstrakte syntax for instruktioner, og måske forklare konverteringen ved hjælp af figurende.

I figurende burde S være I. Og i if else statementet burde det være I_1 og I_2, if og else branchen burde være forskellige after all.

Reply:
Måske burde man også vise at i base casen så bliver instruktionen bare placeret på kanten.

Reply:
Se afsnit 2.2 i formal methods: An app....